src/D2L.Security.OAuth2/Utilities/D2LHttpClient.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace D2L.Security.OAuth2.Utilities
+{
+	public class D2LHttpClient : ID2LHttpClient

```

Comment:
Didn't our strategy pivot to sticking with `HttpClient` and generating the `Task.Run` wrapper?

Reply:
Correct me if I'm wrong @j3parker but I think this is a form of future-proofing so that we can implement true "sync" versions of HttpClient methods in the future. We do have the ability to wrap them in Task.Run but saved that for one-offs and not the HttpClient class

Reply:
I think the interface we're going to need to wrap HTTP is going to look a lot different than what we've got here (like methods like `T GetAsJson<T>( string url );` etc., high level stuff).

I'm OK if we can drop D2LHttpClient for now and rely on the generator, or if that's a hassle I don't care if we have this (just it'll change drastically in a bit, that's all.)

Reply:
It is a bit of a hassle, because given the common names "Get" and "Send", I would have to check the types of whatever is calling them. Anyway, it would have to be a new package so if things are going to change drastically anyway I feel that we shouldn't make additional changes to the generator only to revert them later

Reply:
I guess I'm unsure why we added the `ReadAsStringAsync` wrapping into the generator, when it was directly related to this HttpClient usage - and I thought we were ultimately adding it to avoid having to wrap HttpClient (at the moment, at least).

Whatever works if we're just working incrementally though. I'd rather us not cut a release of D2L.Security.OAuth2 as-is though.

Reply:
I see, it's because ReadAsStringAsync is part of HttpContent not HttpClient which complicates things